### PR TITLE
Use inline flag instead of @inline annotation

### PR DESCRIPTION
**Pull request by odersky**
Convert `@inline` annotations to `inline` flags, not the
other way round as was done before.

This is the first step to fixing #1647, which requires a reorganization how
annotations are handled. 

I wanted to get the irregular treatement of inline annotations out of the way before trying this. 

Review by @smarter. Feel free to cherry-pick for getting the new repo structure to work.

### CODE REVIEW

File: src/dotty/tools/dotc/core/SymDenotations.scala [hunk 1]
```diff
@@ -751,10 +751,7 @@ object SymDenotations {
     //    def isOverridable: Boolean = !!! need to enforce that classes cannot be redefined
     def isSkolem: Boolean = name == nme.SKOLEM
 
-    def isInlineMethod(implicit ctx: Context): Boolean =
-      is(Method, butNot = Accessor) &&
-      !isCompleting &&   // don't force method type; recursive inlines are ignored anyway.
-      hasAnnotation(defn.InlineAnnot)
+    def isInlineMethod(implicit ctx: Context): Boolean = is(InlineMethod, butNot = Accessor)
```
Comment (smarter) [inline src/dotty/tools/dotc/core/SymDenotations.scala]: Is the `butNot` still necessary? Do accessors get the inline flag too somehow?

File: src/dotty/tools/dotc/typer/Namer.scala [hunk 2]
```diff
@@ -521,6 +521,7 @@ class Namer { typer: Typer =>
     mergeCompanionDefs()
     val ctxWithStats = (ctx /: stats) ((ctx, stat) => indexExpanded(stat)(ctx))
     createCompanionLinks(ctxWithStats)
+    //stats foreach enterAnnotations
```
Comment (smarter) [inline src/dotty/tools/dotc/typer/Namer.scala]: Stray comment?

File: src/dotty/tools/dotc/typer/Namer.scala [hunk 3]
```diff
@@ -568,19 +569,9 @@ class Namer { typer: Typer =>
...
-          // create a @inline annotation. Currently, the inlining trigger
-          // is really the annotation, not the flag. This is done so that
-          // we can still compile inline methods from Scala2x. Once we stop
-          // being compatible with Scala2 we should revise the logic to
-          // be based on the flag. Then creating a separate annotation becomes unnecessary.
-          denot.addAnnotation(Annotation(defn.InlineAnnot))
-          addInlineInfo(denot, original)
+          if (cls == defn.InlineAnnot && denot.is(Method, butNot = Accessor))
+            denot.setFlag(Inline)
+          if (denot.isInlineMethod) addInlineInfo(denot, original)
```
Comment (smarter) [inline src/dotty/tools/dotc/typer/Namer.scala]: This means that `addInlineInfo` will be called for every annotation if the method is marked inline, previously `hasInlineAnnot` prevented that.

File: src/dotty/tools/dotc/core/SymDenotations.scala (hunk 1 above)
Comment (odersky) [inline src/dotty/tools/dotc/core/SymDenotations.scala]: They might if we have an accessor of an inline val. Better play it safe.

**Comment by odersky** [comment]
Merging now, because there's more to come...

**Pull request merged**